src/main/java/Models/Report.java

```diff
@@ -0,0 +1,68 @@
+package Models;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+public class Report {
+
+    private final Map<Student, TrainingProgram> studentCourses;
+    private final LocalDateTime reportGenerationTime;
+
+    public Report(Map<Student, TrainingProgram> studentCourses){
+        this.studentCourses = studentCourses;
+        this.reportGenerationTime = LocalDateTime.now();
+    }
+
+    public Report(Map<Student, TrainingProgram> studentCourses, LocalDateTime reportGenerationTime){
+        this.studentCourses = studentCourses;
+        this.reportGenerationTime = reportGenerationTime;
+    }
+
+    public String generateReport(String input){
+        StringBuilder report = new StringBuilder();
+
+        if(input.equals("0") || input.equals("")){
+            studentCourses.forEach((key, value) -> report.append(generateShortReport(key, value)).append("\n\n"));
+        }
+        else {
+            studentCourses.forEach((key, value) -> report.append(generateFullReport(key, value)).append("\n\n"));
+        }
+        return report.toString();
+    }
+
+    private String generateShortReport(Student student, TrainingProgram training) {
+        return String.format("%s (%s) - %s",
+                student.getFullName(),
+                training.getTrainingName(),
+                trainingStatusSupport(training));
+    }
+
+    private String generateFullReport(Student student, TrainingProgram training){
+        return String.format("%s\n%s%s",
+                student.getFullName(),
+                training.toString(),
+                trainingStatusSupport(training));
+    }
+
+    private String trainingStatusSupport(TrainingProgram training){
```

Comment:
one thing that I found just now. this method counts the time to or after completion, but in working hours. It should count in full 24 hour manner